etest_test/build_test.py

```diff
@@ -9,4 +14,193 @@ def test_help() -> None:
+
+    # Check for errors
+    assert result.exit_code == 0
+
+    # Assert all functions have not been called
+    build.assert_not_called()
+    run.assert_not_called()
+    commit.assert_not_called()
+    image_remove.assert_not_called()
+    container_remove.assert_not_called()
+
+
+@pytest.mark.parametrize("arch", ["amd64", "x86", "armv5", "armv6", "armv7", "arm64", "ppc64"])
+def test_base_build(arch: str, mocker: MockerFixture) -> None:
+    """Ensure the base images build correctly."""
+    # Mock qemu management
+    mocker.patch("etest.build.qemu")
+
+    # Mock build functions
+    build = mocker.patch(
+        "etest.build.docker.image.build",
+        return_value=True,
+    )
+    run = mocker.patch(
+        "etest.build.docker.container.run",
+        return_value=True,
+    )
+    commit = mocker.patch("etest.build.docker.container.commit")
+
+    # Mock cleanup functions
+    image_remove = mocker.patch("etest.build.docker.image.remove")
+    container_remove = mocker.patch("etest.build.docker.container.remove")
+
+    # Run the program
+    runner = click.testing.CliRunner()
+    result = runner.invoke(sut.main, ["--arch", arch, "--verbosity", "DEBUG"])
+
+    # Check for errors
+    assert result.exit_code == 0
+
+    # Assert all functions have been called
+    build.assert_called_once()
+    run.assert_called_once()
+    commit.assert_called_once()
+    image_remove.assert_called_once()
+    container_remove.assert_called_once()
+
+
+@pytest.mark.parametrize("libc", ["glibc", "musl", "uclibc"])
+@pytest.mark.parametrize("hardened", ["--hardened", "--no-hardened"])
+def test_alternate_libc_build(libc: str, hardened: str, mocker: MockerFixture) -> None:
+    """Ensure images with alternate libcs build correctly."""
+    # Mock build functions
+    build = mocker.patch(
+        "etest.build.docker.image.build",
+        return_value=True,
+    )
+    run = mocker.patch(
+        "etest.build.docker.container.run",
+        return_value=True,
+    )
+    commit = mocker.patch("etest.build.docker.container.commit")
+
+    # Mock cleanup functions
+    image_remove = mocker.patch("etest.build.docker.image.remove")
+    container_remove = mocker.patch("etest.build.docker.container.remove")
+
+    # Run the program
+    runner = click.testing.CliRunner()
+    result = runner.invoke(sut.main, ["--libc", libc, hardened, "--verbosity", "DEBUG"])
+
+    # Check for errors
     assert result.exit_code == 0
+
+    # Assert all functions have been called
+    build.assert_called_once()
+    run.assert_called_once()
+    commit.assert_called_once()
+    image_remove.assert_called_once()
+    container_remove.assert_called_once()
+
+
+def test_builderror(mocker: MockerFixture) -> None:
+    """Ensure the program fails on a BuildError."""
+    # Raise a ContainerError
+    log = itertools.tee({"error": "mockup"})[0]
+    error = docker.errors.BuildError(reason="Mockup error.", build_log=log)
```

Comment:
If you aren't looking at the reason or the build_log then we can just use the empty values for their types.